cmd/otelcol/main_test.go

```diff
@@ -14,3 +14,111 @@
 // limitations under the License.
 
 package main
+
+import (
+	"context"
+	"net"
+	"os"
+	"runtime"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestRunFromCmdLine(t *testing.T) {
+	tests := []struct {
+		name     string
+		panicMsg string
+		skipMsg  string
+		args     []string
+		timeout  time.Duration
+	}{
+		{
+			name:    "agent",
+			args:    []string{"otelcol", "--config=config/collector/agent_config.yaml"},
+			timeout: 15 * time.Second,
+		},
+		{
+			name:    "gateway",
+			args:    []string{"otelcol", "--config=config/collector/gateway_config.yaml"},
+			timeout: 15 * time.Second,
+		},
+		{
+			name:    "default_discovery",
+			args:    []string{"otelcol", "--discovery", "--config=config/collector/agent_config.yaml"},
+			timeout: 30 * time.Second,
+		},
+		// Running the discovery with --dry-run in CI is not desirable because of the use of os.Exit(0) to end the execution.
```

Comment:
--dry-run was originally intended as a helper/utility  for the field/sales to capture what services are available  but design requirements have changed. I wonder we need this test at all. But we can keep it like this. I do not have strong feelings either way.

Reply:
Got it, I will ask around to see if the need/purpose is clear.